src/data_provider/src/sysInfoLinux.cpp

```diff
@@ -184,14 +185,32 @@ int SysInfo::getCpuCores() const
 int SysInfo::getCpuMHz() const
 {
     int retVal { 0 };
-    std::map<std::string, std::string> systemInfo;
-    getSystemInfo(WM_SYS_CPU_DIR, ":", systemInfo);
+    int cpuFreq { 0 };
+    const auto cpusInfo { Utils::enumerateDir(WM_SYS_CPU_FREC_DIR) };
+    constexpr auto NAME_DIRECCTORY_CPU_FREQ {"^cpu[0-9]+"};
+    const std::regex rex{NAME_DIRECCTORY_CPU_FREQ};
 
-    const auto& it { systemInfo.find("cpu MHz") };
-
-    if (it != systemInfo.end())
+    for (const auto cpu : cpusInfo)
     {
-        retVal = std::stoi(it->second) + 1;
+        std::smatch sm;
+
+        if (regex_search(cpu, sm, rex))
+        {
+            std::fstream file{WM_SYS_CPU_FREC_DIR + cpu + "/cpufreq/cpuinfo_max_freq", std::ios_base::in};
+
+            if (file.is_open())
+            {
+                std::string frequency { };
+
+                std::getline(file, frequency);
+                cpuFreq = std::stoi(frequency);  // Frequency on KHz
+
+                if (cpuFreq / 1000 > retVal)
+                {
+                    retVal = static_cast<int>(cpuFreq / 1000);
+                }
+            }
+        }
     }
 
     return retVal;
```

Comment:
Maybe we could return `retVal / 1000` here and not use ` / 1000` on the previous parts. (Although for that we'd have to change the variable unit to Hz).